src/simple_treeview.rs

```diff
@@ -0,0 +1,98 @@
+//! # TreeView Sample
+//!
+//! This sample demonstrates how to create a TreeView with either a ListStore.
+
+extern crate glib;
+extern crate gtk;
+
+use gtk::prelude::*;
+use gtk::{
+    CellRendererText, Label, ListStore, Orientation, TreeView, TreeViewColumn,
+    Window, WindowPosition, WindowType
+};
+
+fn append_column(tree: &TreeView, id: i32) {
+    let column = TreeViewColumn::new();
+    let cell = CellRendererText::new();
+
+    column.pack_start(&cell, true);
+    // Association of the view's column with the model's `id` column.
+    column.add_attribute(&cell, "text", id);
+    tree.append_column(&column);
+}
+
+fn create_and_fill_model() -> ListStore {
+    // Creation of a model with two rows.
+    let model = ListStore::new(&[u32::static_type(), String::static_type()]);
+
+    // Filling up the tree view.
+    let entries = &["Michel", "Sara", "Liam", "Zelda", "Neo", "Octopus master"];
+    for (i, entry) in entries.iter().enumerate() {
+        model.insert_with_values(None, &[0, 1], &[&(i as u32 + 1), &entry]);
+    }
+    model
+}
+
+fn create_and_setup_view() -> TreeView {
+    // Creating the tree view.
+    let tree = TreeView::new();
+
+    tree.set_headers_visible(false);
+    // Creating the two columns inside the view.
+    append_column(&tree, 0);
+    append_column(&tree, 1);
+    tree
+}
+
+fn main() {
+    if gtk::init().is_err() {
+        println!("Failed to initialize GTK.");
+        return;
+    }
+
+    let window = Window::new(WindowType::Toplevel);
+
+    window.set_title("Simple TreeView example");
+    window.set_position(WindowPosition::Center);
+
+    window.connect_delete_event(|_, _| {
+        gtk::main_quit();
+        Inhibit(false)
+    });
+
+    // Creating a vertical layout to place both tree view and label in the window.
+    let vertical_layout = gtk::Box::new(Orientation::Vertical, 0);
+
+    // Creation of the label.
+    let label = Label::new(None);
+
+    let tree = create_and_setup_view();
+
+    let model = create_and_fill_model();
+    // Setting the model into the view.
+    tree.set_model(Some(&model));
+    // Adding the view to the layout.
+    vertical_layout.add(&tree);
+    // Same goes for the label.
+    vertical_layout.add(&label);
+
+    // When a line is selected, this signal is emitted so we connect to it through this method.
```

Comment:
It's not. But it says what it does, and that's what we're trying to do after all.